### ISSUE TITLE
Recent versions of `vite-node` fail to execute a script when ESM is imported

### ISSUE THREAD

**Issue by antoinerey** [vitest-dev/vitest#1208]
### Describe the bug

**TLDR; `vite-node` higher than `0.8.5` fails to run a script where ESM is imported.**

I'm using `vite-node` to execute some code that relies on several dependencies. One of those dependencies (in my case `msw`) provides both ESM and UMD bundles. When executing the script, `vite-node` imports the ESM bundle, and immediately crashes.

The script works fine when running `vite-node@0.8.5`, but fails with any higher version. I believe that the turning point is the [c4a25ae](https://github.com/vitest-dev/vitest/commit/c4a25ae) commit, where some checks to inline dependencies where removed.

### Reproduction

https://github.com/antoinerey/vite-node-reproduction-esm-import

### System Info

```shell
System:
    OS: macOS 12.0.1
    CPU: (8) x64 Intel(R) Core(TM) i7-8557U CPU @ 1.70GHz
    Memory: 104.10 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 18.0.0 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.6.0 - /usr/local/bin/npm
    Watchman: 2022.03.21.00 - /usr/local/bin/watchman
  Browsers:
    Brave Browser: 100.1.37.116
    Chrome: 101.0.4951.41
    Firefox: 98.0.1
    Safari: 15.1
  npmPackages:
    vite-node: ^0.10.0 => 0.10.0
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
msw doesn't have `"type": "module"` in their package.json (https://unpkg.com/browse/msw@0.39.2/package.json). Failing is correct behaviour. Please, ask them to fix it.

**Comment by antoinerey** [comment]
👍 Will do! Also, thanks for the dual-packaging example. In my opinion, the behaviour around CJS and ESM is quite tricky to get working, and your explanation definitely helps me!

Side question though, I'm trying to understand if the recent changes were expected or not. Since it's working with versions below `0.9.0`, I'm wondering if the behaviour at that time was incorrect. In other words, was it working because `vite-node` was to permissive, and didn't not follow specifications?

**Comment by sheremet-va** [comment]
Previously `vite-node` transformed files that didn't need transformation, because it was a valid ESM code inside `node_modules`. It just assumed that the code will not be a correct ESM, because it is usually wrong in the Node ecosystem :) But at some point we added `isValidESM` check, that follows the specification and more reliable.

We try not to transform code inside `node_modules` for performance reasons.

We have two important options to fix this
- `fallbackCJS: true` will try to find cjs versions of the file and load it instead of rellying on potentially wrong esm output 
- `deps.inline` will just always process files to make them compatible

**Comment by sheremet-va** [comment]
> Since the script does not run through vitest, I believe that everything under the test property in Vite/Vitest configuration is not taken into account, right?

Actually, you are correct! Unfortunately, right now it doesn't support this. But I don't see why we shouldn't allow it. PR welcome :)

**Comment by antoinerey** [comment]
Aaaand, PR is here: https://github.com/vitest-dev/vitest/pull/1215!
